py/torch_tensorrt/dynamo/_refit.py

```diff
@@ -444,33 +473,41 @@ def refit_module_weights(
                     settings=settings,
                     weight_name_map=None,
                 )
+        deallocate_module(new_submodule)
 
         # clear EXCLUDE_WEIGHTS flag
         serialization_config = engine.create_serialization_config()
         serialization_config.clear_flag(trt.SerializationFlag.EXCLUDE_WEIGHTS)
         serialized_engine = engine.serialize_with_config(serialization_config)
 
-        if isinstance(
-            compiled_submodule, (PythonTorchTensorRTModule, TorchTensorRTModule)
-        ):
+        if isinstance(compiled_submodule, PythonTorchTensorRTModule):
+            compiled_submodule.serialized_engine = bytes(serialized_engine)
+        elif isinstance(compiled_submodule, TorchTensorRTModule):
             compiled_submodule.engine = None  # Clear the engine for TorchTensorRTModule, otherwise it won't be updated
             compiled_submodule.serialized_engine = bytes(serialized_engine)
             compiled_submodule.setup_engine()
-
         elif inline_module:
             new_engine_info = list(engine_info)
             new_engine_info[ENGINE_IDX] = bytes(serialized_engine)
             refitted_engine = torch.classes.tensorrt.Engine(tuple(new_engine_info))
             setattr(compiled_module, f"{name}_engine", refitted_engine)
 
+        del engine
+        gc.collect()
+        torch.cuda.empty_cache()
+
+    deallocate_module(new_partitioned_module)
+
     if verify_output and arg_inputs is not None:
+        new_gm.to(torch.cuda.current_device())
```

Comment:
We should ensure we use the device that's passed in via args/Compilation settings or default device and not call rely on torch cuda calls unless it is needed.